_posts/2020-04-28-archivesspace-cleanup-dates.md

```diff
@@ -0,0 +1,85 @@
+|           Date Expression           |    Result    |
+|:-----------------------------------:|:------------:|
+|               No Date               | Does nothing |
+| N.D. [or “n.d.”, “n/d”, ”nd”,,“ND”] | Does nothing |
+|            N/A [or “n/a”]           | Does nothing |
+|               d. 1913               | Does nothing |
+|            Dec. 13, 1979            | Does nothing |
+|             1979 Jan. 12            | Does nothing |
+|             Probably 1938           | Does nothing |
+|           Exhibited: 1960           | Does nothing |
+
+You may be wondering why I added a table that includes an entire row that says “Does nothing” - well this is actually really important! The fact that Timewalk will do nothing when it can’t understand a date expression is actually what we want to happen. Other tools might replace the date expression with “NULL” when they are unable to parse it, which means the original data is now lost.
+
+## Not Exactly a Silver Bullet, But Close Enough!
+Of course no python module, ruby gem, or plug-in would be able to handle some of our really wild date expressions. A chunk of these obscure dates needed to be addressed manually. This manual work took a few days to complete. It included editing certain dates in ArchivesSpace that we knew Timewalk would be unable to parse. By doing some quality control first, we could more confidently rely on Timewalk to do the remainder of the work, ensuring that most of our dates would be parsed.
+
+Additionally, most other expressions that caused Timewalk to fail followed patterns. This meant they could be fixed automatically by a function that essentially would “find and replace” certain patterns in the date expression field. Things like “Jan.” could easily be changed to “January” and “no date” or “n.d.” could be changed to “undated”. We could also easily change “d. 1913” to “1910” by removing any string containing “d.”  and in a similar way, remove “Exhibited: ” from “Exhibited: 1960”.
+
+## Our Solution: A python script that triggers Timewalk
+Working with the Digital Strategies Team, I provided a list of expressions that would need to be changed in the “find and replace” style function. We wanted to take this opportunity to standardize the language used in the date expression field. For example: archival objects with an unknown date should always be “undated” and months should always be spelled out completely (i.e. January, February, March).
+
+We also decided it might be best to take a piecemeal approach, and have the script run on a single finding aid at time instead of the entire repository. Since we would be running the script on the production server, there was a chance we would be updating a finding aid that another archivist was working on, which would cause confusion in ArchivesSpace. By selecting a specific resource id before running the script, we could ensure that we would not interfere with any processing projects that were currently in progress.
```

Comment:
```suggestion
We also decided it might be best to take a piecemeal approach, and have the script run on a single finding aid at a time instead of the entire repository. Since we would be running the script on the production server, there was a chance we would be updating a finding aid that another archivist was working on, which would cause confusion in ArchivesSpace. By selecting a specific resource id before running the script, we could ensure that we would not interfere with any processing projects that were currently in progress.
```